### PR TITLE
"Split" client into objects and core client

### PR DESCRIPTION
**Pull request by kim-cc**
### Description
Create interfaces for core client and objects client, making it clear to users what can be used in each.
Also moved some types to the internal package which is enforced at the compiler level.

### Type(s) of Change (Split the PR if you check many)
- [ ] Added user feature
- [X] Added technical feature
- [ ] Added tests
- [ ] Refactor
- [ ] Bugfix
- [ ] Updated documentation
- [ ] Kubernetes changes
- [ ] CI/CD workflow changes

### Testing Checklist
- [ ] I have added/updated unit tests for functional changes.
- [ ] I have added/updated integration tests for changes that affect dependent systems.
- [ ] I have added/updated end-to-end tests for feature changes.

### General Checklist
- [X] I have pulled in the latest changes from master.
- [X] I have run `make lint`.
- [X] I have successfully run `make tests`.
- [ ] I have updated relevant CI/CD workflows.
- [ ] I have updated relevant Kubernetes manifests/scripts.
- [ ] I have updated/added relevant (internal and external) documentation (readme, doc comments, etc).
- [ ] I have updated the dependency map to reflect my changes.
- [ ] I have added the license to new source code files.
- [X] I have spent some time looking over the full diff before creating this PR.

### CODE REVIEW

File: pkg/client/client.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
+package client
```
Comment (philip-cc) [inline pkg/client/client.go:1]: Couldn't this and `def.go` just be in the root folder? It makes the import statement a little nicer for consumers of the client.

Comment (philip-cc) [inline pkg/client/client.go:1]: There should probably be some copyright stuff here.

File: pkg/client/client.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+func NewCoreClient(ctx context.Context, endpoint, certPath string) (CoreClient, error) {
+	client, err := internal.NewClient(ctx, endpoint, certPath)
+	if err != nil {
+		return nil, err
+	}
+
+	return client, nil
+}
+
+// NewObjectsClient creates a new Encryptonize Core client. Note that in order to call endpoints that require
```
Comment (philip-cc) [inline pkg/client/client.go:76]:
```suggestion
// NewObjectsClient creates a new Encryptonize Objects client. Note that in order to call endpoints that require
```

File: pkg/client/client.go (hunk 1 above)
Comment (kim-cc) [reply pkg/client/client.go:1]: Restructuring made to fix this

Comment (kim-cc) [reply pkg/client/client.go:1]: Fixed

**Review by philip-cc — APPROVED** [review]

**Pull request merged**